src/main.js

```diff
@@ -0,0 +1,87 @@
+const myLibrary = [];
+
+function Book(author, title, pages, read) {
+  this.author = author;
+  this.title = title;
+  this.pages = pages;
+  this.read = read;
+}
+function setStatus(read, button) {
+  if (read) {
+    button.innerText = 'Read';
+  } else {
+    button.innerText = 'Not read yet';
+  }
+}
+
+function switchStatus(e) {
+  const index = e.srcElement.attributes['1'].value;
+  myLibrary[parseInt(index, 10)].read = !myLibrary[parseInt(index, 10)].read;
+  setStatus(myLibrary[parseInt(index, 10)].read, e.target);
+}
+
+function removeBook(e) {
+  const body = document.querySelector('.flex');
+  const child = e.target.parentNode;
+  if (child && myLibrary.length !== 0) {
+    body.removeChild(child);
+    const index = e.srcElement.attributes['1'].value;
+    delete myLibrary[index];
+  }
+}
+
+function render() {
+  const div = document.createElement('div');
+  const title = document.createElement('h3');
+  const author = document.createElement('p');
+  const pages = document.createElement('p');
+  const button = document.createElement('button');
+  button.addEventListener('click', (e) => {
+    switchStatus(e);
+  });
+  const remove = document.createElement('button');
+  remove.addEventListener('click', (e) => {
+    removeBook(e);
+  });
+  const body = document.querySelector('.flex');
+  myLibrary.forEach((item, i) => {
+    title.innerHTML = `Title: ${item.title}`;
+    author.innerHTML = `Author: ${item.author}`;
+    pages.innerHTML = `Number of pages: ${item.pages}`;
+    button.innerHTML = 'Status';
+    button.setAttribute('class', 'status');
+    remove.innerHTML = 'Remove';
+    remove.setAttribute('class', 'remove');
+    div.appendChild(title).setAttribute('class', 'title');
+    div.appendChild(author).setAttribute('class', 'author');
+    div.appendChild(pages).setAttribute('class', 'pages');
+    div.appendChild(button).setAttribute('rel', i);
+    div.appendChild(remove).setAttribute('rel', i);
+
+    body.appendChild(div).setAttribute('class', 'book-container');
+  });
+}
+
+function addBookToLibrary() {
+  const author = document.getElementsByName('author')[0].value;
+  const title = document.getElementsByName('title')[0].value;
+  const pages = document.getElementsByName('pages')[0].value;
+  const read = false;
+  const book = new Book(
+    this.author = author,
+    this.title = title,
+    this.pages = pages,
+    this.read = read,
+  );
+  myLibrary.push(book);
+  document.getElementsByName('author')[0].value = '';
+  document.getElementsByName('title')[0].value = '';
+  document.getElementsByName('pages')[0].value = '';
+  render();
+}
+
+(function add() {
+  document.getElementById('submit').addEventListener('click', () => {
+    addBookToLibrary();
+  });
+}());
```

Comment:
I'm curious about why you are doing this in an IIFE? 🤔 I don't think there is a need for that here, you can just add the event listener without the IIFE.

Reply:
Hey Mauricio, We use IIFE to avoid global scope in our script.

Reply:
Ok, perfect 🤗, make sense, although you are not declaring variables inside the IIFE, just setting an event listener so the global scope is not polluted either way.

Reply:
Yes, you're right, just good practices. 😁

Reply:
Oh, got it! That's very good practice, move everything inside the IIFE, everything is on the global scope right now. To properly avoid having everything on the global scope move all the code inside the IIFE, something like this:

```js
(() => {
  const myLibrary = [];
  // ...
  // all the rest of your code
  //...
  document.getElementById('submit').addEventListener('click', () => {
    addBookToLibrary();
  });
}());
```